src/helpers/wallet/index.js

```diff
@@ -0,0 +1,48 @@
+import Web3 from 'web3'
+import { SUPPORTED_CHAINS } from '../../data/constants'
+import { networks } from '../../data/networks'
+import { createWalletModal } from './modal'
+
+let connection
+
+export const connect = async () => {
+  const modal       = createWalletModal()
+  const provider    = await modal.requestProvider()
+  const web3        = new Web3(provider)
+  const chainId     = await web3.eth.getChainId()
+  const [ address ] = await web3.eth.getAccounts()
+
+  connection = { modal, provider, web3 }
+
+  return { address, chainId, modal, provider, web3 }
+}
+
+export const disconnect = async () => {
+  if (! connection) return
+
+  const { modal, provider } = connection
+
+  if (provider?.close)
+    await provider.close()
+
+  if (modal.clearCachedProvider)
+    await modal.clearCachedProvider()
```

Comment:
I personally prefer this style, but since on some other places it is used with curly braces, let's keep it that way for the time being.
```suggestion
  if (provider?.close) {
    await provider.close()
  }

  if (modal.clearCachedProvider) {
    await modal.clearCachedProvider()
  }
```